src/main/dynamo/Beyonce.ts

```diff
@@ -159,14 +159,24 @@ export class Beyonce {
     return await batchWriteItems({ table: this.table, client: this.client, putItems: maybeEncryptedItems, deleteItems })
   }
 
-  /** Perform N Dynamo operations in an atomic transaction */
+  /**
+   * Perform N Dynamo operations in an atomic transaction.
+   *
+   * @param params.putItems Set `mustBeUnique` to true on a record within putItems to cancel the transaction if a
+   *  record with the same partition key and sort key (if applicable) already exists.
+   */
   async batchWriteWithTransaction<T extends TaggedModel>(params: {
     clientRequestToken?: string
-    putItems?: T[]
+    putItems?: (T & { mustBeUnique?: boolean })[]
```

Comment:
@jcarver989 and I had discussed an interface like the following:

```
await beyonce.batchWriteWithTransaction({ putItems: [{ ...item, conditionExpression: (item) => item.mustBeUnique() }] })
```

Since that is more complex but accomplishes the same goal, I'd prefer to go with the approach in this PR unless reviewers strongly prefer another approach.

Reply:
Supporting the full range of condition expressions would be a nice thing to eventually support, but I'm fine with having just a boolean option to start. 

re: naming `failIfNotUnique` might communicate the resultant behavior more clearly, but up to you on which  name you prefer.